src/BSON/Document.stub.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @generate-class-entries static
+ * @generate-function-entries static
+ */
+
+namespace MongoDB\BSON;
+
+final class Document implements \IteratorAggregate
+{
+    private function __construct() {}
+
+    final static public function fromBSONString(string $bson): Document {}
+
+    final static public function fromJSON(string $json): Document {}
+
+#if PHP_VERSION_ID >= 80000
+    final static public function fromPHP(mixed $value): Document {}
+#else
+    /** @param mixed $value */
+    final static public function fromPHP($value): Document {}
+#endif
+
+#if PHP_VERSION_ID >= 80000
+    final public function get(string $key): mixed {}
+#else
+    /** @return mixed */
+    final public function get(string $key) {}
+#endif
+
+    final public function getIterator(): Iterator {}
+
+    final public function has(string $key): bool {}
+
+#if PHP_VERSION_ID >= 80000
+    final public function toPHP(?array $typeMap = null): mixed {}
```

Comment:
Should this be `array|object` (and in the `@return` annotation below)? That's what you used in the stub for the `toPHP` function.

Also, is there a particular reason the stub for `src/BSON/functions.c` is one directory up in `src/functions.stub.php`?

Reply:
Another note regarding the location of the function stub file:
Currently, the `gen_stub.php` script only generates a single file containing an `ext_functions` variable containing all function entries. This probably works well for PHP core since namespaces aren't widely used there, but for our usages it presents a few problems. When creating stub files, I created the functions stub file in the root and included functions from all namespaces (`MongoDB\BSON` and `MongoDB\Driver\Monitoring`) in the file.

Reply:
Oh, I didn't notice that _all_ functions were in there. That makes sense. Thanks for confirming.